relay-kafka/src/lib.rs

```diff
@@ -0,0 +1,54 @@
+pub mod config;
+#[cfg(feature = "processing")]
+mod producer;
+#[cfg(feature = "processing")]
+mod utils;
+
+use failure::Fail;
+
+#[cfg(feature = "processing")]
+pub use producer::*;
+
+/// Kafka producer errors.
+#[cfg(feature = "processing")]
+#[derive(Fail, Debug)]
+pub enum ProducerError {
+    #[fail(display = "failed to send kafka message")]
+    SendFailed(#[cause] rdkafka::error::KafkaError),
+    #[fail(display = "failed to create kafka producer: invalid kafka config")]
+    InvalidConfig(#[cause] rdkafka::error::KafkaError),
+    #[fail(display = "failed to serialize kafka message")]
+    InvalidMsgPack(#[cause] rmp_serde::encode::Error),
+    #[fail(display = "failed to serialize json message")]
+    InvalidJson(#[cause] serde_json::Error),
+    #[fail(display = "invalid kafka shard")]
+    InvalidShard,
+}
+
+/// Kafka configuration errors.
+#[derive(Fail, Debug)]
+pub enum ConfigError {
```

Comment:
Is there a reason this is not in `config`? This causes some strange imports now where most types get imported from `Config` except for the related error.

Reply:
yeah, I will reorganize the modules and make it simpler and uniform.